expression/column.go

```diff
@@ -199,6 +199,8 @@ type Column struct {
 
 	hashcode []byte
 
+	// IsInsertSelect means if this column is the select field of insertSelect
+	IsInsertSelect bool
```

Comment:
This change looks too ad-hoc, we are trying very hard now to remove unnecessary fields from `Column`, do we have any other solution which is more graceful to achieve the purpose of this PR? For example, add a flag parameter in `FindColumn` to indicate if "ambiguous column" error should be ignored? @winoros Any suggestion?

Reply:
This is not easy to do.If two colums both come from `select` of `insert ... select...`,we can ignore the `ambiguous error` .But if one comes from `select` and another comes from `insert`. we must catche the error.
For example:
1. ` insert into t(a,b,f1) select f1,f2,f2 from t2 on duplicate key update a = f2` 
2. ` insert into t(a,b,f1) select f1,f2,f2 from t2 on duplicate key update a = f1`
The first one is ok, but second one is not.
So we must know each column if it is from `select` of `insert... select`.

Reply:
@jingyugao For you first case. I constructed another SQL
```
insert into t3(a, b, c) select f2 f1, f2 f1,f3 f1 from t2 src on duplicate key update a = f1+100;
```
Which `f1` in `a=f1+100` points to?